lib/upload.js

```diff
@@ -81,41 +82,41 @@ class Upload {
       return;
     }
 
-    getSource(file, this.options.chunkSize, (err, source) => {
-      if (err) {
-        this._emitError(err);
-        return;
-      }
+    if (this._source) {
+      this._start(this._source);
```

Comment:
Would you mind explaining the rational behind this if branch? It looks a bit suspicious to me. Is it about retries?

Reply:
Yes this is needed for retries. Previously this was written as something like `this._source ||= getSource(...`. Now that `getSource` is async, it's a little more verbose.

When a retry occurs `.start()` is called again. The uploader instance has the current offset and will request a chunk at that point from the source. However, if the source is a newly instantiated `StreamSource`, it doesn't know the current offset. We need to reuse the source so it saves any bytes in its buffer and knows the byte offset at the start of that buffer.

Reply:
Ok, that makes sense. Thank you